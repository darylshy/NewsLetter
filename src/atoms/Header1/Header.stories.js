import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';
import { withKnobs, text } from '@storybook/addon-knobs/react';
import React from 'react';

import Header1 from './Header1';

storiesOf('Atoms/Header', module)
  .addDecorator(withInfo)
  .addDecorator(withKnobs)
  .add('Standard h1 header', () => <Header1 header={text('header')} />, {
    info: {
      text: 'H1 Header Element'
    }
  });
