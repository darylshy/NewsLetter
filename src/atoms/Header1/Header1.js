import React from 'react';
import PropTypes from 'prop-types';
import classes from './Header1.module.scss';

const Header1 = props => {
  return <h1 className={classes.Header1}>{props.header.toUpperCase()}</h1>;
};

Header1.propTypes = {
  header: PropTypes.string.isRequired
};

export default Header1;
