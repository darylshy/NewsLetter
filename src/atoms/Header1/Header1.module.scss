@import "../../styles/global/globals.scss";
.Header1 {
  font-family: "Azo-Sans-Bold";
  margin: 0;
  @include desktop {
    font-size: 2.4rem;
  }
  @include tabletL {
    font-size: 2rem;
  }
  @include tabletP {
    font-size: 2rem;
  }
  @include mobile {
    font-size: 2rem;
  }
}
