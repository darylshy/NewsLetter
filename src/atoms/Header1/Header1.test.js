import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Header1 from './Header1';
import React from 'react';

configure({ adapter: new Adapter() });

describe('<Header1 />', () => {
  let component;
  beforeEach(() => {
    component = shallow(<Header1 header="header" />);
  });
  it('should render an Header1 component', () => {
    expect(component).toMatchSnapshot();
  });
  it('should find a Header1', () => {
    expect(component.find('h1')).toHaveLength(1);
  });
});
