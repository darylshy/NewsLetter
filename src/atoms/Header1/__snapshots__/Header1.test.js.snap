// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header1 /> should render an Header1 component 1`] = `ShallowWrapper {}`;
